### ISSUE TITLE
Ajouter le support des commandes personnalisée proposé par Discord

### ISSUE THREAD

**Issue by SirMishaa** [NotANameServer/discord#23]
#### Description détaillée de votre suggestion
Mettre à jour le bot pour supporter le support des commandes personnalisée proposé par Discord 
https://discord.com/developers/docs/interactions/slash-commands

NB: J'ai déjà proposé ça sur Discord, mais je l'ajoute ici afin de garder une trace et d'avoir l'avancée de l'implémentation. 

#### Raison·s derrière cette suggestion
Cela permet de donner une idée de toutes les commandes disponibles, en plus des arguments et d'avoir une validation, le bot a de plus en plus de commande et parfois, c'est difficile de toutes les avoir en tête. 

#### Contrainte·s éventuelle·s
Implémentation sur le bot ?

#### Annexe·s / Fichier·s
![5002338abeaf8532983a1be594a10683](https://user-images.githubusercontent.com/42717232/116890409-df9c4b80-ac2d-11eb-9e7a-e231bae18999.png)

**Comment by Mesteery** [comment]
Cette issue devrait plutôt être sur https://github.com/DigitalPulseSoftware/NotaBot.

**Comment by AntoineJT** [comment]
Il ne risque pas d'y avoir un conflit entre bots sur les serveurs où plusieurs bots sont utilisés conjointement ?
Pour connaître les commandes disponibles il suffit de faire !help, c'est potentiellement même plus rapide que de taper des slash commandes au hasard en espérant trouver celle qu'on recherche.

**Comment by SirLynix** [comment]
Les slashs commands sont catégorisés par bot, normalement pas trop de soucis à ce niveau-là

**Comment by SirMishaa** [comment]
> 
> 
> Il ne risque pas d'y avoir un conflit entre bots sur les serveurs où plusieurs bots sont utilisés conjointement ?
> Pour connaître les commandes disponibles il suffit de faire !help, c'est potentiellement même plus rapide que de taper des slash commandes au hasard en espérant trouver celle qu'on recherche.

Non, cela ajoute bien plus qu'un aperçu des commandes disponibles, comme de l'autocomplétion, la validation automatique, une description et diverses informations. 

Autant utiliser le support natif que propose Discord qu'utiliser ``!help`` qui est, en plus, moins pratique pour les utilisateurs.

**Comment by Elanis** [comment]
Après, rien n'empêche d'avoir les deux comportements: le `/` discord, et une commande de help, vu que cette dernière existe déjà iirc.

**Comment by Equinox4** [comment]
> Les slashs commands sont catégorisés par bot, normalement pas trop de soucis à ce niveau-là
Comment on fais avec les slash commands pour choisir tel ou tel bot ?

**Comment by SirMishaa** [comment]
@SirLynix Up ?

**Comment by SirLynix** [comment]
Il faut que je fasse des tests, mais travailler sur le bot n'est pas une de mes priorités du moment (néanmoins ça va bientôt le redevenir).

**Comment by Elanis** [comment]
En ajout de tout ca, je pense que faire un `/guideline` permettant de lister les guidelines et d'en selectionner une pour la prompt serait interessant

**Comment by Equinox4** [comment]
je passe l'état de cette issue à "accepté" mais on ne peut toujours pas le mettre en place tant que discordia v3 n'est pas sorti.
